src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -349,65 +377,105 @@ protected void setTargetIndex(int newTargetIdx)
             return;
         }
 
+        // We need this for sending the FIR. No need to waste cycles if we don't
+        // have it.
+        MediaStream sourceStream
+            = sourceTrack.getMediaStreamTrackReceiver().getStream();
+        if (sourceStream == null)
+        {
+            return;
+        }
+
         RTPEncodingDesc[] sourceEncodings = sourceTrack.getRTPEncodings();
+        if (ArrayUtils.isNullOrEmpty(sourceEncodings))
+        {
+            return;
+        }
 
         int currentTL0Idx = bitstreamController.getCurrentIndex();
+        boolean currentTL0IsActive;
         if (currentTL0Idx > SUSPENDED_INDEX)
         {
             currentTL0Idx
                 = sourceEncodings[currentTL0Idx].getBaseLayer().getIndex();
+            currentTL0IsActive
+                = sourceEncodings[currentTL0Idx].isActive(nowMs);
+        }
+        else
+        {
+            currentTL0IsActive = false;
         }
 
         int targetTL0Idx
-            = sourceEncodings[newTargetIdx].getBaseLayer().getIndex();
+            = sourceEncodings[targetIdx].getBaseLayer().getIndex();
```

Comment:
maybe it's hard to tell because of the diff, but this was intentional to change this from `newTargetIdx` to `targetIdx`?

Reply:
The code in `maybeRequestKeyFrame()` was part of the `setTargetIndex(int newTargetIndex)` method. The new method doesn't have access to the `newTargetIndex` variable, so we use a new variable called `targetIndex` (and we get the value from the bitstreamController).

Reply:
tl;dr; it's the same value, only the name changed